File: Data Structures/Stack/stack.cpp
#include<iostream>
using namespace std;

class Stack{
	int top;
	int size;
	int arr[100];

public:
	Stack(){
		top = 0;
		size = 0;
	}

	void push(int x){
		if(size == 100){
			cout << "Stack is Full..!!" << endl;
		}
		else{
			arr[top] = x;
			top++;
			size++;
			cout << x << " Pushed to Stack..!!" << endl;
		}
	}

	int pop(){
		if(size == 0){
			cout << "Stack is Empty..!!" << endl;
		}
		else{
			top--;
			size--;
			int temp = arr[top];
			cout << temp << " is Popped..!!" << endl;
			return temp;
		}
	}

	void display(){
		for(int i = 0; i < size; i++){
			cout << arr[i] << " ";
		}
		cout << endl;
	}
};

int main(){
	Stack s;
	s.push(1);
	s.push(2);
	s.push(3);
	s.push(4);
	s.push(5);
	s.push(6);
	s.display();
	s.pop();
	s.pop();
	s.pop();
	s.pop();
	s.display();
	return 0;
}

File: Data Structures/Stack/Stack.java
public class Stack{

    // Definition of Stack class 

    int current_size;
    int max_size;
    int top;
    int s[];

    public Stack(int size){
        this.max_size = size;
        this.current_size = 0;
        this.top = 0;
        this.s = new int [size];
    }

    public void push(int value){
        if(this.current_size == this.max_size)
            System.out.println("Stack is full..!!");
        else{
            this.s[top] = value;
            System.out.println(value + " Pushed to Stack..!!");
            top++;
            current_size++;
        }
    }

    public int pop(){
        if(this.current_size == 0){
            System.out.println("Stack is Empty..!!");
            return -1;
        }
        else{
            int item;
            this.top--;
            this.current_size--;
            item = this.s[top];
            System.out.println(item + " is Popped..!!");
            return item;
        }
    }


    public void display(){
     for(int i=0; i<this.current_size; i++){
        System.out.print(this.s[i] + " ");
     }
     System.out.println();
    }

    // Main Function
    public static void main(String args[]) {
        Stack s = new Stack(100);  // Stack of max_size 100 is created
        s.push(1);
	    s.push(2);
	    s.push(3);
	    s.push(4);
	    s.push(5);
        s.push(6);
	    s.display();
	    s.pop();
	    s.pop();
	    s.pop();
	    s.pop();
	    s.display();
    }
}